src/library/basesqltablemodel.cpp

```diff
@@ -32,10 +34,10 @@ BaseSqlTableModel::BaseSqlTableModel(QObject* pParent,
           m_database(pTrackCollection->getDatabase()),
           m_previewDeckGroup(PlayerManager::groupForPreviewDeck(0)),
           m_iPreviewDeckTrackId(-1),
-          m_currentSearch("") {
-    m_bInitialized = false;
-    m_iSortColumn = 0;
-    m_eSortOrder = Qt::AscendingOrder;
+          m_bInitialized(false),
+          m_currentSearch(""),
+          m_trackSourceSortColumn(kIdColumn),
```

Comment:
Yea, BaseTrackCache requires column 0 to be the ID column due to the skipping logic. But the ID column in the table column list might not be the first entry.